codes/classical/bits/repetition code.yml

```diff
@@ -0,0 +1,90 @@
+
+# Use UTF-8 unicode encoding
+# AMS-TeX commands are rendered inside \( ... \) using MathJaX.
+# Allowed external bibliographic references are
+#    \cite{arXiv:#.#} or \cite{arXiv:quant-ph/#},
+#    \cite{doi:#}, or, as a last resort
+#    \cite{manual:{(enter citation line incl. author and year here)}}
+# External websites such as code tables, coding theory packages, github pages linked as
+#    \url{https://example.com/example}
+#    \href{https://example.com/example}{link text}
+# Internal references to codes are 
+#    \hyperref[code:code_id]{link text}
+# Delete instructional comments when submitting
+
+# code id, physical, logical are all lower case
+code_id: repetition
+physical: bits
+logical: bits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title is singular
+name: 'Repetition code'
+
+# First reference with the code (optional).
+introduced: '\cite{doi:10.1070/RM1997v052n06ABEH002155}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'This code encodes 1-bit information into a \(n\)-bit string. \(n\) needs to be an odd number, since receiver will pick the majority at decoding stage. Its basic idea is to increase the code distance by repeating the logical information several times. It is a \((n,1)\)-Hamming code.'
+
+protection: 'Protects against bit-flip and erasure up to \(\frac{n-1}{2}\)-bit error. Bit-flip error and erasure error are the common in classical communication processes.'
+
+# This field starts a list of specific labeled subfields; do not leave it
+# empty. If empty, comment out. Also, indentations are important!
+features:
+
+  # The fields below relate to specific properties of the code that are worth
+  # noting. In some cases (see below), there might be multiple pieces of
+  # information associated with a field (e.g., mutliple encoders/decoders,
+  # multiple facts about fault tolerance, etc.); in such cases, the field
+  # accepts a list of text entries, with one item per piece of information.
+  #
+  # Any fields below should be omitted if they do not apply.
+
+  rate: 'The code rate is \(\frac{1}{n}\). The code distance is \(n\). Its generator matrix is \(G=\left[\begin{smallmatrix}1 & 1&\cdots& 1 & 1 \end{smallmatrix}\right]\). '
+
+  # This field that starts a list of strings, indexed by a dash; do not leave it
+  # empty. If empty, comment out.
+  encoders: 
+    - 'The encoding process is \(1\rightarrow 111...1\), \(0\rightarrow 000...0\). Repeating the information \(n\)-times.'
+
+  decoders:
+    - 'Calculate the Hamming weight \(d_H\) of the code. If \(d_H\leq \frac{n-1}{2}\), decode the code as 0. If \(d_H\geq \frac{n+1}{2}\), decode the code as 1.'
+
+
+  threshold: 
+    - 'Suppose each bit has probability \(p\) of being received correctly, independent for each bit. The probability that a repetition code is received correctly is \(\sum_{k=0}^{(n-1)/2}\frac{n!}{k!(n-k)!}p^{n-k}(1-p)^{k}\).'
```

Comment:
I thought threshold is usually something for quantum codes. This looks like the beginning of a threshold calculation for this classical code. Is there some large-n limit that one can take here? If you are aware of a standard calculation here, then feel free to reproduce. Otherwise, I would not spend too much time for this.